www/src/components/layout/WithApplicationUpdate.js

```diff
@@ -1,48 +1,55 @@
-import { useCallback, useContext, useState } from 'react'
+import {
+  useCallback, useContext, useEffect, useState,
+} from 'react'
 
-import * as serviceWorker from '../../serviceWorker'
+import * as serviceWorker from '../../serviceWorkerRegistration'
 
 import { PluralConfigurationContext } from '../login/CurrentUser'
 
 const COMMIT_KEY = 'git-commit'
 
 export const DEFAULT_COMMIT = 'plural-default-commit'
 
-export const getCommit = () => localStorage.getItem(COMMIT_KEY) || DEFAULT_COMMIT
-export const setCommit = sha => localStorage.setItem(COMMIT_KEY, sha)
+export const getCommit = () => sessionStorage.getItem(COMMIT_KEY) || DEFAULT_COMMIT
+export const setCommit = sha => sessionStorage.setItem(COMMIT_KEY, sha)
 
 function WithApplicationUpdate({ children }) {
-  const [open, setOpen] = useState(true)
+  const [time, setTime] = useState(Date.now())
   const config = useContext(PluralConfigurationContext)
   const commit = getCommit()
-
+  const [stale, setStale] = useState(commit !== config.gitCommit)
   const reloadApplication = useCallback(() => {
-    console.log('reloading')
-    if (process.env.NODE_ENV === 'production') {
-      const promise = serviceWorker.unregister() || Promise.resolve('done')
-
-      promise.then(() => {
-        setCommit(config.gitCommit)
-        window.location.reload()
-      })
-    }
-    else {
+    const promise = serviceWorker.unregister() || Promise.resolve('done')
+
+    setStale(false)
+
+    promise.then(() => {
       setCommit(config.gitCommit)
-      setOpen(false)
-    }
-  }, [config, setOpen])
+      window.location.reload()
+    })
+  }, [config])
+  const reloadOnStale = () => (stale ? reloadApplication() : null)
+
+  // force rerender every 60s to check if version hasn't changed
+  useEffect(() => {
+    const interval = setInterval(() => setTime(Date.now()), 60000)
+
+    setStale(commit !== config.gitCommit)
 
-  const stale = commit !== config.gitCommit
+    return () => clearInterval(interval)
+  }, [time, setStale, commit, config])
 
-  if (!stale || !open) return children({ reloadApplication, shouldReloadApplication: false })
+  // Run only once on first app load
+  // It enforces full app reload if app has been updated
+  useEffect(() => {
+    reloadOnStale()
+  }, [])
```

Comment:
React Hook useEffect has a missing dependency: 'reloadOnStale'. Either include it or remove the dependency array.
GitHub is right here.

Reply:
Including a dependency will trigger this effect on every dependency change and this is exactly what I wanted to avoid here. Right now it is only executed once on application load.